src/System.Collections.Concurrent/tests/BlockingCollectionTests.cs

```diff
@@ -48,5 +48,1984 @@ public static void TestBlockingCollectionBasic()
 
             Task.WaitAll(tks);
         }
+
+        /// <summary>
+        /// Bug description: BlockingCollection throws  InvalidOperationException when calling CompleteAdding even after adding and taking all elements
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        [OuterLoop]
+        public static void RunBlockingCollectionTest_BugFix544259()
+        {
+            int count = 8;
+            CountdownEvent cde = new CountdownEvent(count);
+            BlockingCollection<object> bc = new BlockingCollection<object>();
+
+            //creates 8 consumers, each calling take to block itself
+            for (int i = 0; i < count; i++)
+            {
+                int myi = i;
+                Task.Run(() =>
+                {
+                    bc.Take();
+                    cde.Signal();
+                });
+            }
+            //create 8 producers, each calling add to unblock a consumer
+            for (int i = 0; i < count; i++)
+            {
+                int myi = i;
+                Task.Run(() =>
+                {
+                    bc.Add(new object());
+                });
+            }
+
+            //CountdownEvent waits till all consumers are unblocked
+            cde.Wait();
+            bc.CompleteAdding();
+        }
+
+        // as part of the bugfix for 626345, this code was suffering occassional ObjectDisposedExceptions due
```

Comment:
Try to remove internal bug numbers, they aren't going to help us track anything here, and I wouldn't be surprised if this bug number was irrelevant to us now anyways